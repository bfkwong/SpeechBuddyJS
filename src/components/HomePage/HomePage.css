.flexContainer {
  display: flex;
  flex-direction: column;
  justify-content: center;
  text-align: center;
  margin: auto;
}

.flexItem {
  padding: 5px;
  margin-top: 10px;
}

#stars {
  display: block;
  max-width: 100px;
  max-height: 100px;
  width: auto;
  height: auto;
  margin: auto;
}

.main {
  animation: fadein ease 2s;
}

.overview {
  animation: fadein ease 3s;
}

.features {
  animation: fadein ease 4s;
  background-image: linear-gradient(-257deg, #a3ffe5, #a3faff 50%, #a3d7ff);
}

.reviews {
  align-content: center;
  animation: fadein ease 5s;
}

@keyframes fadein {
  from {
    opacity: 0;
  }
  to {
    opacity: 1;
  }
}

ul {
  list-style: none;
}

p {
  padding: 2em;
  margin: auto;
}
