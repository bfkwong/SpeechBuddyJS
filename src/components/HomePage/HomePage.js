import React from "react";
import { Link } from "react-router-dom";
import Stars from "../../Images/Stars.png";
import "./HomePage.css";

export default function HomePage() {
  return (
    <div className="flexContainer">
      <header className="header">
        <div className="main flexItem">
          <h1>Welcome to SpeechBuddy</h1>
          <h5>A Free Text and Speech Analyzer</h5>
          <h6>For Students and Teachers Alike</h6>
        </div>
        <div className="overview flexItem">
          <h2>Speech Buddy Overview</h2>
          <p>
            Speech Buddy is a text and speech analysis toolkit for quick
            statistics and visualizations on any given text.
          </p>
        </div>
      </header>
      <header className="features flexItem">
        <h2> Analyze Text Tone & Word Variance for Single Text Speeches </h2>
        <div>
          <p>
            Check the tone of your text before you perform in front of an
            audience. Ensure that your audience reacts the way you expect
            depending on your intended tone:
          </p>
          <ul>
            <li>
              <Link to="/SpeechBuddyJS/single_text">Single Text Analysis</Link>
            </li>
          </ul>
        </div>
        <h2> Plagiarism and Similarity Comparison Between 2+ Text Speeches </h2>
        <div>
          <p>Compare two or more .txt files for speech similarity:</p>
          <ul>
            <li>
              <Link to="/SpeechBuddyJS/two_txt_plag">Two Text Analysis</Link>
            </li>
            <li>
              <Link to="/SpeechBuddyJS/multi_text">Multi Text Analysis</Link>
            </li>
          </ul>
        </div>
      </header>
      <header className="reviews flexItem">
        <h2>Satisfied User Reviews</h2>
        <div>
          <img src={Stars} alt="5Stars" id="stars" />
          <p>
            "SpeechBuddy is wonderful! I used to have to manually check each
            essay for plagiarism but now that it's automated I don't have to! I
            give it a 9/9 for quality tools and a 1/1 for aesthetic. Total
            10/10!"
          </p>
          <span>- Karen J. (High School Teacher)</span>
          <img src={Stars} alt="5Stars" id="stars" />
          <p>
            "This was a very helpful tool when writing speeches for my Speech
            Writing class. The sentiment and toxicity analysis helps me better
            gauge how my audience would feel during my presentation! Definitely
            recommend utilizing before a big presentation!"
          </p>
          <span>- Smar T. (Harvard Student)</span>
          <img src={Stars} alt="5Stars" id="stars" />
          <p>"My dad like this alot."</p>
          <span>- Smar T. Jr. (Preschool Student)</span>
        </div>
      </header>
    </div>
  );
}
