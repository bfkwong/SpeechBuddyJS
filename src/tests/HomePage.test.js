import React from "react";
import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import HomePage from "../components/HomePage/HomePage";
import renderer from "react-test-renderer";
import { unmountComponentAtNode } from "react-dom";

describe("HomePage", () => {
  let container;

  beforeEach(() => {
    container = document.createElement("div");
    document.body.appendChild(container);
  });

  afterEach(() => {
    if (container.hasChildNodes()) {
      unmountComponentAtNode(container);
    }
    container.remove();
    container = null;
  });

  test("HomePage renders without failure", () => {
    render(
      <BrowserRouter>
        <HomePage />
      </BrowserRouter>,
      container
    );
  });

  test("HomePage snapshot test", () => {
    const appTree = renderer
      .create(
        <BrowserRouter>
          <HomePage />
        </BrowserRouter>
      )
      .toJSON();
    expect(appTree).toMatchSnapshot();
  });
});
