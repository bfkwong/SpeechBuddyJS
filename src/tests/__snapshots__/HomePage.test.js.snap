// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage HomePage snapshot test 1`] = `
<div
  className="flexContainer"
>
  <header
    className="header"
  >
    <div
      className="main flexItem"
    >
      <h1>
        Welcome to SpeechBuddy
      </h1>
      <h5>
        A Free Text and Speech Analyzer
      </h5>
      <h6>
        For Students and Teachers Alike
      </h6>
    </div>
    <div
      className="overview flexItem"
    >
      <h2>
        Speech Buddy Overview
      </h2>
      <p>
        Speech Buddy is a text and speech analysis toolkit for quick statistics and visualizations on any given text.
      </p>
    </div>
  </header>
  <header
    className="features flexItem"
  >
    <h2>
       Analyze Text Tone & Word Variance for Single Text Speeches 
    </h2>
    <div>
      <p>
        Check the tone of your text before you perform in front of an audience. Ensure that your audience reacts the way you expect depending on your intended tone:
      </p>
      <ul>
        <li>
          <a
            href="/SpeechBuddyJS/single_text"
            onClick={[Function]}
          >
            Single Text Analysis
          </a>
        </li>
      </ul>
    </div>
    <h2>
       Plagiarism and Similarity Comparison Between 2+ Text Speeches 
    </h2>
    <div>
      <p>
        Compare two or more .txt files for speech similarity:
      </p>
      <ul>
        <li>
          <a
            href="/SpeechBuddyJS/two_txt_plag"
            onClick={[Function]}
          >
            Two Text Analysis
          </a>
        </li>
        <li>
          <a
            href="/SpeechBuddyJS/multi_text"
            onClick={[Function]}
          >
            Multi Text Analysis
          </a>
        </li>
      </ul>
    </div>
  </header>
  <header
    className="reviews flexItem"
  >
    <h2>
      Satisfied User Reviews
    </h2>
    <div>
      <img
        alt="5Stars"
        id="stars"
        src="Stars.png"
      />
      <p>
        "SpeechBuddy is wonderful! I used to have to manually check each essay for plagiarism but now that it's automated I don't have to! I give it a 9/9 for quality tools and a 1/1 for aesthetic. Total 10/10!"
      </p>
      <span>
        - Karen J. (High School Teacher)
      </span>
      <img
        alt="5Stars"
        id="stars"
        src="Stars.png"
      />
      <p>
        "This was a very helpful tool when writing speeches for my Speech Writing class. The sentiment and toxicity analysis helps me better gauge how my audience would feel during my presentation! Definitely recommend utilizing before a big presentation!"
      </p>
      <span>
        - Smar T. (Harvard Student)
      </span>
      <img
        alt="5Stars"
        id="stars"
        src="Stars.png"
      />
      <p>
        "My dad like this alot."
      </p>
      <span>
        - Smar T. Jr. (Preschool Student)
      </span>
    </div>
  </header>
</div>
`;
